doc/manual/Checker_Library.md

```diff
@@ -0,0 +1,84 @@
+<!---
+This Source Code Form is subject to the terms of the Mozilla
+Public License, v. 2.0. If a copy of the MPL was not distributed
+with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+-->
+
+# Checker Library
+
+## Introduction
+
+A Checker Library is a collection of modules for one domain. Typically, it
+contains a set of checks for one standard or to test the input file for one
+specific Use Case or in one specific environment. The checks may be distributed
+in multiple Checker Bundles in the Checker Library.
+
+Each Checker Library needs to have a documentation what is included. Here we
+have to distinguish between "Checks" and "Rules". A standard specify rules,
+which are described in textual form. A check is a test implementation which
+inspects the input document to verify if it is compliant to the standard.
+
+## Rules
+
+### Rule Characteristics
+
+- Rules are atomic, easy understandable and independent of how many rules are
+  necessary in the standard.
+- A Rule should be part of a standard, as it is today in many ASAM standards.
+- Rules can come from ASAM, but also from other organizations or can be company
+  internal rules.
+- A Checker Library may also need to document rules, if they are not explicitly
+  written in the according standard. This documentation can be used as a proposal
+  to the standardization project to improve the standard itself.
+
+### Rule Documentation Template
+
+If a Checker Library needs to define and document own rules, then the following
+template can be used.
+
+**Rule ID:** _The Rule ID should not change, if there is no change in the rule._
+
+**Description:** _Clear textual description of the rule._
+
+**Severity:** _mandatory or recommendation_
```

Comment:
If a recommendation rule is not verified by the checker, we get a warning.

If a mandatory rule is not verified by the checker, we get an error.

Is there a use case in which to verify a rule certain pre-conditions shall be met during verification? (e.g., other rule must be checked beforehand; e.g., some necessary conditions must exists in the checked file in order to make a rule mandatory). And if it is the case, is it a change in its severity or should we consider some additional description field?

To be completely honest, we were not able to provide a valid use-case on OpenDRIVE standard. However, since we are not competent in other standards covered by the framework we decided to raise the question anyway.

Reply:
I've also thinking about your point. We definitely have situations where a second check only make sense if a first one was passed without any issues. Don't know if this could/should be implemented with proper effort or if this should only be documented somewhere/somehow.

To be discussed...

Reply:
@MatteoRagni : We decided that dependency handling should only be done within a Checker Library. So, we only need to consider here in the PR, if `**Severity:** _mandatory or recommendation_` is enough info (at the moment).

What do you think?

Reply:
@andreaskern74: given the context, I would go with a paragraph that tries to explain what we mean with severity, more than the actual list of severity levels. Something like

```suggestion
**Severity:** indicates the rule's importance level as asserted by the standard (e.g., respecting the rule is mandatory or a recommendation)
```

---

**Confused Side Note**: I'm not fully confident about this description, since it is missing the concept of _the current context of the current file is changing the severity level of this rule_. 

Anyway, **I think I would leave this information out of this document**, since it is not clear if this is something that actually happens (i.e., I cannot provide an example for the OpenDRIVE standard. The only examples that comes in my mind are the fact that a rule may enable or disable other rules, but this does not affect the severity).